packages/fast-components-react-base/src/toolbar/toolbar.spec.tsx

```diff
@@ -0,0 +1,249 @@
+                <div role="menuitem">two</div>
+            </Toolbar>
+        );
+
+        rendered.childAt(0).simulate("keydown", { keyCode: keyCodeArrowUp });
+
+        expect(rendered.state("focusIndex")).toBe(0);
+    });
+
+    test("should not move focus up when the up arrow is pressed in horizontal mode", (): void => {
+        const rendered: any = mount(
+            <Toolbar orientation={Orientation.horizontal} initialFocusIndex={1}>
+                <div role="menuitem">one</div>
+                <div role="menuitem">two</div>
+            </Toolbar>
+        );
+
+        rendered.childAt(0).simulate("keydown", { keyCode: keyCodeArrowUp });
+
+        expect(rendered.state("focusIndex")).toBe(1);
+    });
+
+    test("should move focus left when the left arrow is pressed in horizontal mode", (): void => {
+        const rendered: any = mount(
+            <Toolbar orientation={Orientation.horizontal} initialFocusIndex={1}>
+                <div role="menuitem">one</div>
+                <div role="menuitem">two</div>
+            </Toolbar>
+        );
+
+        rendered.childAt(0).simulate("keydown", { keyCode: keyCodeArrowLeft });
+
+        expect(rendered.state("focusIndex")).toBe(0);
+    });
+
+    test("should not move focus left when the right arrow is pressed in vertical mode", (): void => {
+        const rendered: any = mount(
+            <Toolbar orientation={Orientation.vertical} initialFocusIndex={1}>
+                <div role="menuitem">one</div>
+                <div role="menuitem">two</div>
+            </Toolbar>
+        );
+
+        rendered.childAt(0).simulate("keydown", { keyCode: keyCodeArrowLeft });
+
+        expect(rendered.state("focusIndex")).toBe(1);
+    });
+
+    test("should place focus any child with the proper role", (): void => {
+        const children: React.ReactNode[] = Toolbar.DefaultFocusableRoles.map(
+            (key: string): React.ReactNode => {
+                return (
+                    <button key={key} role={key}>
+                        {key}
+                    </button>
+                );
+            }
+        );
+
+        const rendered: any = mount(<Toolbar>{children}</Toolbar>);
+
+        expect(rendered.state("focusIndex")).toBe(0);
+
+        const itemCount: number = Toolbar.DefaultFocusableRoles.length - 1;
+        for (let i: number = 0; i < itemCount; i++) {
+            rendered.childAt(0).simulate("keydown", { keyCode: keyCodeArrowRight });
+            expect(rendered.state("focusIndex")).toBe(i + 1);
+        }
+    });
+
+    test("should not place focus any child without a the proper role", (): void => {
+        const rendered: any = mount(
+            <Toolbar>
+                <button />
+                <button role="menuitem">two</button>
+                <button />
+                <button role="menuitem">three</button>
+                <button />
+                <button role="menuitem">four</button>
```

Comment:
Should this be role based? Is it possible that you could have an element with an implicit role that should receive focus?

Reply:
next version not role based.